modules/compute/main.tf

```
@@ -0,0 +1,64 @@
+resource "azurerm_network_interface" "network_interface" {
+  name                = "${var.vm_name}-nic"
+  location            = var.rg_location
+  resource_group_name = var.rg_name
+
+  ip_configuration {
+    name                          = "testconfiguration1"
+    subnet_id                     = var.subnet_id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+resource "azurerm_virtual_machine" "vm" {
```

```suggestion
resource "azurerm_linux_virtual_machine" "vm" {
```
Потрібно використовувати ресурс azurerm_linux_virtual_machine

---

fixed